src/GraphQl/Type/TypeBuilder.php

```diff
@@ -224,25 +224,24 @@ public function getNodeInterface(): InterfaceType
     public function getResourcePaginatedCollectionType(GraphQLType $resourceType, string $resourceClass, string $operationName): GraphQLType
     {
         $shortName = $resourceType->name;
+        $paginationType = $this->pagination->getGraphQlPaginationType($resourceClass, $operationName);
 
-        if ($this->typesContainer->has("{$shortName}Connection")) {
-            return $this->typesContainer->get("{$shortName}Connection");
+        if ($this->typesContainer->has("{$shortName}Connection_{$paginationType}")) {
+            return $this->typesContainer->get("{$shortName}Connection_{$paginationType}");
         }
 
-        $paginationType = $this->pagination->getGraphQlPaginationType($resourceClass, $operationName);
-
         $fields = 'cursor' === $paginationType ?
             $this->getCursorBasedPaginationFields($resourceType) :
             $this->getPageBasedPaginationFields($resourceType);
 
         $configuration = [
-            'name' => "{$shortName}Connection",
-            'description' => "Connection for $shortName.",
+            'name' => "{$shortName}Connection_{$paginationType}",
+            'description' => "Connection for $shortName. ({$paginationType})",
             'fields' => $fields,
         ];
 
         $resourcePaginatedCollectionType = new ObjectType($configuration);
-        $this->typesContainer->set("{$shortName}Connection", $resourcePaginatedCollectionType);
+        $this->typesContainer->set("{$shortName}Connection_{$paginationType}", $resourcePaginatedCollectionType);
```

Comment:
IMO, the type should be:
`shortNameCursorConnection`
Could you please make the changes to obtain this?

Reply:
Sure, done.